acceptance: unflake Elixir tests

otan opened the pull request:
Elixir tests previously relied on downloading dependencies at test time,
rather than being pre-built with the Docker container. As such, change
the Docker container to contain the pre-build dependencies and the test
should run assuming it is installed.

Resolves #52339

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52341)
<!-- Reviewable:end -->

otan committed:
acceptance: unflake Elixir tests

Elixir tests previously relied on downloading dependencies at test time,
rather than being pre-built with the Docker container. As such, change
the Docker container to contain the pre-build dependencies and the test
should run assuming it is installed.

Release note: None

otan commented:
Closing this because Elixir upgraded from 1.10.2 to 1.10.4, and suddenly you can't even connect to the DB:

```

20:44:51.361 [error] Postgrex.Protocol (#PID<0.975.0>) failed to connect: ** (DBConnection.ConnectionError) ssl connect: TLS client: In state wait_finished at tls_connection.erl:1323 generated CLIENT ALERT: Fatal - Handshake Failure
 malformed_handshake_data - {:tls_alert, {:handshake_failure, 'TLS client: In state wait_finished at tls_connection.erl:1323 generated CLIENT ALERT: Fatal - Handshake Failure\n malformed_handshake_data'}}


  1) test handles decimal correctly (Cockroach.TestDecimal)
     test/decimal_test.exs:4
     ** (DBConnection.ConnectionError) connection not available because of disconnection
     code: for dec <- [
     stacktrace:
       (db_connection 1.1.3) lib/db_connection.ex:934: DBConnection.checkout/2
       (db_connection 1.1.3) lib/db_connection.ex:750: DBConnection.run/3
       (db_connection 1.1.3) lib/db_connection.ex:1141: DBConnection.run_meter/3
       (db_connection 1.1.3) lib/db_connection.ex:592: DBConnection.prepare_execute/4
       (postgrex 0.13.5) lib/postgrex.ex:146: Postgrex.query/4
       test/decimal_test.exs:57: anonymous fn/3 in Cockroach.TestDecimal."test handles decimal correctly"/1
       (elixir 1.10.4) lib/enum.ex:2111: Enum."-reduce/3-lists^foldl/2-0-"/3
       test/decimal_test.exs:17: (test)
```

Seems like an arcane build related issue, I don't think it's worth the effort on this considering we already have unit tests to test the same thing.
